src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -227,16 +196,17 @@ public GlobalSearchBar(LibraryTabContainer tabContainer,
 
         Timer searchTask = FxTimer.create(Duration.ofMillis(SEARCH_DELAY), this::updateSearchQuery);
         BindingsHelper.bindBidirectional(
-                searchQueryProperty,
+                stateManager.activeSearchQuery(searchType),
                 searchField.textProperty(),
                 searchTerm -> {
                     // Async update
                     searchTask.restart();
                 },
                 query -> setSearchTerm(query.map(SearchQuery::getQuery).orElse("")));
 
-        this.searchQueryProperty.addListener((obs, oldValue, newValue) -> newValue.ifPresent(this::updateSearchResultsForQuery));
-        this.stateManager.activeDatabaseProperty().addListener(obs -> searchQueryProperty.get().ifPresent(this::updateSearchResultsForQuery));
+        stateManager.searchResultSize(searchType).addListener((obs, oldVal, newVal) ->
```

Comment:
This listener will not be fired when the search matches all entries. (The label that shows the number of matched entries will not shown in that case).
This happens because the size of the filtered list in `MainTableDataModel` before and after the search will be the same (All entries).
https://github.com/LoayGhreeb/jabref/blob/e13e6a260fc111662346e8f1c0e6d6e5ddd37304/src/main/java/org/jabref/gui/maintable/MainTableDataModel.java#L53-L59